lib/rule-context.js

```diff
@@ -92,12 +92,14 @@ function mergeFixes(fixes, sourceCode) {
     const start = fixes[0].range[0];
     const end = fixes[fixes.length - 1].range[1];
     let text = "";
-    let lastPos = start;
+    let lastPos = Math.max(0, start);
 
     for (const fix of fixes) {
-        assert(fix.range[0] >= lastPos, "Fix objects must not be overlapped in a report.");
+        assert(fix.range[0] >= lastPos || fix.range[0] < 0, "Fix objects must not be overlapped in a report.");
```

Comment:
Would this fail to detect a conflict if two fixes have ranges `[-1, 0]` and `[-1, 5]`? It seems like the assertion will always pass for fixes that start at `-1`.